numpy/typing/_callable.py

```diff
@@ -0,0 +1,140 @@
+        # Note that `__other: bool_` is absent here
+        @overload  # platform dependent
+        def __call__(self, __other: int) -> Union[int32, int64]: ...
+        @overload
+        def __call__(self, __other: float) -> float64: ...
+        @overload
+        def __call__(self, __other: complex) -> complex128: ...
+        @overload
+        def __call__(self, __other: _NumberType) -> _NumberType: ...
+
+    class _BoolTrueDiv(Protocol):
+        @overload
+        def __call__(self, __other: _BoolLike) -> float64: ...
+        @overload  # platform dependent
+        def __call__(self, __other: int) -> Union[float32, float64]: ...
+        @overload
+        def __call__(self, __other: float) -> float64: ...
+        @overload
+        def __call__(self, __other: complex) -> complex128: ...
+        @overload
+        def __call__(self, __other: integer) -> floating: ...
+        @overload
+        def __call__(self, __other: _NumberType) -> _NumberType: ...
+
+    class _TD64Div(Protocol[_NumberType_co]):
+        @overload
+        def __call__(self, __other: timedelta64) -> _NumberType_co: ...
+        @overload
+        def __call__(self, __other: _FloatLike) -> timedelta64: ...
+
+    class _IntTrueDiv(Protocol):
+        @overload
+        def __call__(self, __other: Union[int, float, bool_, integer]) -> floating: ...
+        @overload
+        def __call__(self, __other: complex) -> complexfloating[floating]: ...
+
+    class _UnsignedIntArithmetic(Protocol):
+        @overload
+        def __call__(self, __other: Union[bool, unsignedinteger]) -> unsignedinteger: ...
+        @overload
+        def __call__(self, __other: Union[int, signedinteger]) -> Union[signedinteger, floating]: ...
+        @overload
+        def __call__(self, __other: float) -> floating: ...
+        @overload
+        def __call__(self, __other: complex) -> complexfloating[floating]: ...
+
+    class _SignedIntArithmetic(Protocol):
+        @overload
+        def __call__(self, __other: Union[int, signedinteger]) -> signedinteger: ...
+        @overload
+        def __call__(self, __other: float) -> floating: ...
+        @overload
+        def __call__(self, __other: complex) -> complexfloating[floating]: ...
+
+    class _FloatArithmetic(Protocol):
+        @overload
+        def __call__(self, __other: _FloatLike) -> floating: ...
+        @overload
+        def __call__(self, __other: complex) -> complexfloating[floating]: ...
+
+    class _ComplexArithmetic(Protocol):
+        def __call__(self, __other: _ComplexLike) -> complexfloating[floating]: ...
+
+    class _NumberArithmetic(Protocol):
+        def __call__(self, __other: _NumberLike) -> number: ...
+
+else:
+    _BoolArithmetic = Any
+    _BoolSub = Any
+    _BoolTrueDiv = Any
+    _TD64Div = Any
+    _IntTrueDiv = Any
+    _UnsignedIntArithmetic = Any
+    _SignedIntArithmetic = Any
+    _FloatArithmetic = Any
+    _ComplexArithmetic = Any
+    _NumberArithmetic = Any
```

Comment:
I think these names would be better  with an `Op` or `Method` suffix

Reply:
Nice and short; I like it.

Reply:
Fixed in https://github.com/numpy/numpy/pull/17273/commits/9a9ce45adc99cfec5e9342508d3adf83cab42e78.